### PR TITLE
Add e2e test to simulate control plane node reboot

### PR DESCRIPTION
**Pull request by jluhrsen**
These tests simply kills the control plane pods to simulate the
reboot scenario.

This test does not really simulate the reboot, because:
 1) Kind cluster's etcd stores data in-memory and it looses that data
    after the reboot, so we skip deleting this pod.
 2) Deleting the ovs-node pod will tear down the pod networking and
    and the restart does not build it back, so we skip deleting this
    pod as well.

this is work picked up from an old PR by Anil:
  https://github.com/ovn-org/ovn-kubernetes/pull/1891

Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

File: test/e2e/e2e.go [hunk 1]
```diff
@@ -584,6 +606,74 @@ var _ = ginkgo.Describe("e2e control plane", func() {
...
+		ovnKubeMasterNode, err := findOvnKubeMasterNode()
+		framework.ExpectNoError(err, fmt.Sprintf("unable to find current master of ovnkuber-master cluster %v", err))
+
+		podChan, errChan := make(chan *v1.Pod), make(chan error)
+		go func() {
+			defer ginkgo.GinkgoRecover()
+			checkContinuousConnectivity(f, "", "connectivity-test-continuous", extDNSIP, 53, 30, 30, podChan, errChan)
+		}()
+
+		testPod := <-podChan
```
Comment (trozet) [inline test/e2e/e2e.go:621]: i think you need to read from the errChan first before you read from the podChan according to the checkContinuousConnectivity text

File: test/e2e/e2e.go [hunk 2]
```diff
@@ -584,6 +606,74 @@ var _ = ginkgo.Describe("e2e control plane", func() {
...
+	ginkgo.FIt("should provide Internet connection continuously when all ovnkube-master pods are killed.", func() {
+		ginkgo.By(fmt.Sprintf("Running container which tries to connect to %s in a loop", extDNSIP))
+
+		podChan, errChan := make(chan *v1.Pod), make(chan error)
+		go func() {
+			defer ginkgo.GinkgoRecover()
+			checkContinuousConnectivity(f, "", "connectivity-test-continuous", extDNSIP, 53, 30, 30, podChan, errChan)
+		}()
+
+		testPod := <-podChan
```
Comment (trozet) [inline test/e2e/e2e.go:656]: same here

File: test/e2e/e2e.go [hunk 3]
```diff
@@ -481,6 +482,25 @@ func restartOVNKubeNodePod(clientset kubernetes.Interface, namespace string, nod
...
+	annotationJson := make(map[string]interface{})
+	if err = json.Unmarshal([]byte(annotation), &annotationJson); err != nil {
+		return "", fmt.Errorf("could not parse annotation %q", annotation)
+	}
+	ovnkubeMasterNode, ok := annotationJson["holderIdentity"]
+	if !ok {
+		framework.Failf("holderIdentity not found in the configmap (ovn-kubernetes-master) annotation")
+		return "", fmt.Errorf("holderIdentity not found in the configmap (ovn-kubernetes-master) annotation")
+	}
+	framework.Logf(fmt.Sprintf("master instance of ovnkuber-master is running on node %s", ovnkubeMasterNode))
```
Comment (trozet) [inline test/e2e/e2e.go:500]: nit: ovnkube-master

**Review by trozet — CHANGES_REQUESTED** [review]

File: test/e2e/e2e.go [hunk 4]
```diff
@@ -584,6 +606,80 @@ var _ = ginkgo.Describe("e2e control plane", func() {
 		framework.ExpectNoError(err, "one or more nodes failed to go back ready, schedulable, and untainted")
 	})
 
+	ginkgo.FIt("should provide Internet connection continuously when all pods are killed on node running master instance of ovnkube-master.", func() {
```
Comment (martinkennelly) [inline test/e2e/e2e.go:609]:
```suggestion
	ginkgo.It("should provide Internet connection continuously when all pods are killed on node running master instance of ovnkube-master.", func() {
```
Didnt review the rest of this but just noticed this.

**Review by trozet — APPROVED** [review]

**Pull request merged**